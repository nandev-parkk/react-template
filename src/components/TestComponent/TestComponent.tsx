interface TestComponentProps {
	name: string;
	age: number;
}

export default function TestComponent({ name, age }: TestComponentProps) {
	return (
		<div>
			<span>{name}</span>
			<span>{age}</span>
		</div>
	);
}
