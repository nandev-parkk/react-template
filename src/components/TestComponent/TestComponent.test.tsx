import TestComponent from './TestComponent';
import { render } from '@testing-library/react';

describe('<TestComponent />', () => {
	it('matches snapshot', () => {
		const utils = render(<TestComponent name='shpark' age={20} />);
		expect(utils.container).toMatchSnapshot();
	});
	it('props가 맞는지 확인', () => {
		const utils = render(<TestComponent name='shpark' age={20} />);
		utils.getByText('shpark');
		utils.getByText(20);
	});
});
