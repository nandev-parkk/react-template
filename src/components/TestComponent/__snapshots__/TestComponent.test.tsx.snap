// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestComponent /> matches snapshot 1`] = `
<div>
  <div>
    <span>
      shpark
    </span>
    <span>
      20
    </span>
  </div>
</div>
`;
